website_sale_product_minimal_price/controllers/main.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2020 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+from odoo import http, tools
+from odoo.http import request
+from odoo import fields
+
+
+class WebsiteSaleProductDetailAttributeImage(WebsiteSale):
+
+    @http.route(['/sale/get_combination_info_pricelist_atributes'],
+                type='json',
+                auth="public",
+                website=True)
+    def get_combination_info_pricelist_atributes(
+            self, product_id, actual_qty, **kwargs):
+        """Special route to use website logic in get_combination_info override.
+        This route is called in JS by appending _website to the base route.
+        """
+        actual_qty = int(actual_qty)
+        product = request.env["product.product"].search([("id", "=", product_id)])
+        current_website = request.env["website"].get_current_website()
+        today = fields.Date.today()
+        res = []
+        pricelist = current_website.get_current_pricelist()
+        pricelist_variant_items = pricelist.item_ids.filtered(
+            lambda i: i.product_id.id == product.id
+            and (not i.date_start or i.date_start <= today)
+            and (not i.date_end or today <= i.date_end))
+        if pricelist_variant_items:
+            item_min_qty = min(pricelist_variant_items, key=lambda i: i.min_quantity)
+            min_qty = item_min_qty.min_quantity
+            pricelist_tmpl_items = pricelist.item_ids.filtered(
+                lambda i: i.product_tmpl_id.id == product.product_tmpl_id.id
+                and i.min_quantity < min_qty
+                and (not i.date_start or i.date_start <= today)
+                and (not i.date_end or today <= i.date_end))
+        else:
+            pricelist_tmpl_items = pricelist.item_ids.filtered(
+                lambda i: i.product_tmpl_id.id == product.product_tmpl_id.id
+                and (not i.date_start or i.date_start <= today)
+                and (not i.date_end or today <= i.date_end))
+        self._prepare_dictionary(pricelist_variant_items, product, res)
+        self._prepare_dictionary(pricelist_tmpl_items, product, res)
+        res.sort(key=lambda i: i['min_qty'])
+        # We use the 3 as a limit of data at the frontend, because if we have a
+        # lot of entries at the pricelist item_ids it could be uncomfortable for the
+        # user.
+        if len(res) <= 3:
+            return res
+        else:
+            pointer = 0
+            for el in range(len(res)):
+                if res[el]['min_qty'] > actual_qty:
+                    break
+                pointer = el
+            if pointer == len(res) - 1 or pointer == len(res) - 2:
+                pointer = len(res) - 3
+            return res[pointer:pointer + 3]
```

Comment:
I think that here you can use:
```suggestion
        candidates = list(filter(lambda x: x['min_qty']<=actual_qty, res))
        return candidates[:3]
```

Reply:
If I do that we can't see the offers of the product. You can see what happen on next gif
![1](https://user-images.githubusercontent.com/35952655/94789744-9ead3180-03d5-11eb-9450-45b0cf04cf87.gif)